buildSrc/src/main/groovy/io/vantiq/extsrc/assy/tasks/AssemblyGen.groovy

```diff
@@ -0,0 +1,422 @@
+    public static final String CAMEL_RUNTIME_ROUTE_DOCUMENT = 'routeDocument'
+    public static final String YAML_ROUTE_SUFFIX = '.routes.yaml'
+
+    public final Project project
+    
+    @Inject
+    AssemblyGen(Project project) {
+        this.project = project
+        if (project == null) {
+            throw new GradleException('AssemblyGen requires a non-null project.')
+        }
+    }
+    
+    @OutputDirectory
+    File getOutputDirectory() {
+        if (!(project.getBuildDir() instanceof File)) {
+            throw new GradleException('Internal Error: buildDir not a file: ' + project.getBuildDir())
+        }
+    
+        Path buildDir = project.getBuildDir().toPath()
+        assert buildDir != null
+        return Path.of(buildDir.toAbsolutePath().toString(), KAMELET_GEN_BASE).toFile()
+    }
+    
+    /**
+     * Overall processing for the generation of assemblies from kamelet declarations.
+     *
+     * @throws Exception
+     */
+    @TaskAction
+    void generateAssemblies() throws Exception {
+        Path buildDir = project.getBuildDir()?.toPath()
+        assert buildDir != null
+        if (!(Files.exists(buildDir) && Files.isDirectory(buildDir))) {
+            throw new GradleException('Internal Error: BuildDir not present or not directory: ' + buildDir)
+        }
+        Path generateBase = Paths.get(buildDir.toAbsolutePath().toString(), KAMELET_GEN_BASE)
+        if (!Files.exists(generateBase)) {
+            log.info('Creating base dir: {}', generateBase)
+            generateBase = Files.createDirectories(generateBase)
+        }
+        LoadSettings settings = LoadSettings.builder().build()
+        Load load = new Load(settings)
+    
+        // From the gradle data, construct the list of resolved dependencies & build a class loader to use
+        Configuration conf = project.getConfigurations().getByName('runtimeClasspath')
+        Set<File> files = conf.resolve()
+        log.info('File count from runtimeClasspath: {}', files.size())
+        File[] jars = files.toArray(new File[0])
+        ArrayList<URL> urls = new ArrayList<>(jars.length)
+        for (File f: jars) {
+            log.debug('Adding {} to list of URLs', f.toURI().toURL())
+            urls.add(f.toURI().toURL())
+        }
+        URLClassLoader classloader =
+                new URLClassLoader(urls.toArray(new URL[0]) as URL[])
+        // Find the URL that contains the 'kamelets/' folder (which contains the kamelets to process)
+        URL klUrl = classloader.getResource(KAMELETS_RESOURCE_FOLDER)
+        log.debug('URL for kamelets: {}', klUrl)
+        if (klUrl == null) {
+            throw new GradleException("kamelets' resource is null")
+        }
+        
+        // Find path for jar file containing 'kamelets' folder
+        String klString = klUrl.toString()
+        klString = klString.substring(0, klString.indexOf('!/'))
+        klString = klString.substring(9)
+        def jff = new JarFile(klString)
```

Comment:
_try-with-resources_ replaced with `.withCloseable` due to groovy support/conversion